src/container/rpc.h

```diff
@@ -468,49 +472,558 @@ CRT_RPC_DECLARE(cont_tgt_epoch_aggregate, DAOS_ISEQ_CONT_TGT_EPOCH_AGGREGATE,
 		DAOS_OSEQ_CONT_TGT_SNAPSHOT_NOTIFY)
 
+/* FIXME: cont_prop_set vs cont_prop_set_v8 needed (different inputs) */
+
 #define DAOS_ISEQ_CONT_PROP_SET	/* input fields */		 \
 	((struct cont_op_in)	(cpsi_op)		CRT_VAR) \
 	((daos_prop_t)		(cpsi_prop)		CRT_PTR) \
 	((uuid_t)		(cpsi_pool_uuid)	CRT_VAR)
 
+#define DAOS_ISEQ_CONT_PROP_SET_V8 /* input fields */                                              \
+	((struct cont_op_v8_in)(cpsi_op)CRT_VAR)((daos_prop_t)(cpsi_prop)CRT_PTR)(                 \
+	    (uuid_t)(cpsi_pool_uuid)CRT_VAR)
+
 #define DAOS_OSEQ_CONT_PROP_SET	/* output fields */		 \
 	((struct cont_op_out)	(cpso_op)		CRT_VAR)
 
 CRT_RPC_DECLARE(cont_prop_set, DAOS_ISEQ_CONT_PROP_SET, DAOS_OSEQ_CONT_PROP_SET)
+CRT_RPC_DECLARE(cont_prop_set_v8, DAOS_ISEQ_CONT_PROP_SET_V8, DAOS_OSEQ_CONT_PROP_SET)
+
+/* FIXME: cont_acl_update vs cont_acl_update_v8 needed (different inputs) */
 
 #define DAOS_ISEQ_CONT_ACL_UPDATE	/* input fields */	 \
 	((struct cont_op_in)	(caui_op)		CRT_VAR) \
 	((struct daos_acl)	(caui_acl)		CRT_PTR)
 
+#define DAOS_ISEQ_CONT_ACL_UPDATE_V8 /* input fields */                                            \
+	((struct cont_op_v8_in)(caui_op)CRT_VAR)((struct daos_acl)(caui_acl)CRT_PTR)
+
 #define DAOS_OSEQ_CONT_ACL_UPDATE	/* output fields */	 \
 	((struct cont_op_out)	(cauo_op)		CRT_VAR)
 
-CRT_RPC_DECLARE(cont_acl_update, DAOS_ISEQ_CONT_ACL_UPDATE,
-		DAOS_OSEQ_CONT_ACL_UPDATE)
+CRT_RPC_DECLARE(cont_acl_update, DAOS_ISEQ_CONT_ACL_UPDATE, DAOS_OSEQ_CONT_ACL_UPDATE)
+CRT_RPC_DECLARE(cont_acl_update_v8, DAOS_ISEQ_CONT_ACL_UPDATE_V8, DAOS_OSEQ_CONT_ACL_UPDATE)
 
 #define DAOS_ISEQ_CONT_ACL_DELETE	/* input fields */	 \
 	((struct cont_op_in)	(cadi_op)		CRT_VAR) \
 	((d_string_t)		(cadi_principal_name)	CRT_VAR) \
 	((uint8_t)		(cadi_principal_type)	CRT_VAR)
 
+#define DAOS_ISEQ_CONT_ACL_DELETE_V8 /* input fields */                                            \
+	((struct cont_op_v8_in)(cadi_op)CRT_VAR)((d_string_t)(cadi_principal_name)CRT_VAR)(        \
+	    (uint8_t)(cadi_principal_type)CRT_VAR)
+
 #define DAOS_OSEQ_CONT_ACL_DELETE	/* output fields */	 \
 	((struct cont_op_out)	(cado_op)		CRT_VAR)
 
-CRT_RPC_DECLARE(cont_acl_delete, DAOS_ISEQ_CONT_ACL_DELETE,
-		DAOS_OSEQ_CONT_ACL_DELETE)
+CRT_RPC_DECLARE(cont_acl_delete, DAOS_ISEQ_CONT_ACL_DELETE, DAOS_OSEQ_CONT_ACL_DELETE)
+CRT_RPC_DECLARE(cont_acl_delete_v8, DAOS_ISEQ_CONT_ACL_DELETE_V8, DAOS_OSEQ_CONT_ACL_DELETE)
 
 static inline int
-cont_req_create(crt_context_t crt_ctx, crt_endpoint_t *tgt_ep, crt_opcode_t opc,
-		crt_rpc_t **req)
+cont_req_create(crt_context_t crt_ctx, crt_endpoint_t *tgt_ep, crt_opcode_t opc, uuid_t ci_pool_hdl,
+		uuid_t ci_uuid, uuid_t ci_hdl, uint64_t *req_timep, crt_rpc_t **req)
 {
-	crt_opcode_t opcode;
-
-	opcode = DAOS_RPC_OPCODE(opc, DAOS_CONT_MODULE, dc_cont_proto_version ?
-				 dc_cont_proto_version : DAOS_CONT_VERSION);
+	int                    rc;
+	crt_opcode_t           opcode;
+	static __thread uuid_t cli_id;
```

Comment:
[Question] Every thread of every libdaos application needs to allocate this in addition to `dti_uuid` and our other per-thread static variables, correct? Would it make sense to combine this with `dti_uuid`, for example, by adding a `common` API function `daos_get_client_uuid` that gets (and initializes lazily) `dti_uuid`?

Reply:
Thanks.